### PR TITLE
Some in place refactoring must mostly comments as TODOs

### PR DESCRIPTION
**Pull request by JamesJeffryes**
@scanon I figured that ToDo's in a PR was as good a way to discuss ideas as any. I'll keep updating the PR as we discuss

### CODE REVIEW

File: IndexRunner/EventUtils.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import logging
 
 
+# TODO: Should this just be a FileHandler in in the logger
```
Comment (jayrbolton) [inline IndexRunner/EventUtils.py:11]: for reference https://docs.python.org/3/howto/logging-cookbook.html#using-logging-in-multiple-modules

File: IndexRunner/IndexerUtils.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ def __init__(self, config):
             token = os.environ.get('KB_AUTH_TOKEN')
         self.mr = MethodRunner(config, token=token)
         self.ep = EventProducer(config)
+        # TODO: shouldn't we uniformly use YAML?
```
Comment (jayrbolton) [inline IndexRunner/IndexerUtils.py]: :+1:

File: IndexRunner/IndexerUtils.py [hunk 3]
```diff
@@ -174,6 +178,7 @@ def _get_id(self, upa):
         """
         Return the elastic id
         """
+        # TODO: Should this be stricter?
```
Comment (jayrbolton) [inline IndexRunner/IndexerUtils.py]: :+1: `re.match(r'^\d+\/\d+\/\d+$', 'xyz')`

File: IndexRunner/IndexerUtils.py [hunk 4]
```diff
@@ -222,13 +227,8 @@ def _get_ws_info(self, wsid):
...
         meta = info[8]
         # Don't index temporary narratives
-        temp = False
-        if meta.get('is_temporary') == 'true':
-            temp = True
-
-        public = False
-        if info[6] != 'n':
-            public = True
+        temp = (meta.get('is_temporary') == 'true')
```
Comment (jayrbolton) [inline IndexRunner/IndexerUtils.py:230]: dont need parens

File: IndexRunner/IndexerUtils.py [hunk 5]
```diff
@@ -350,10 +355,10 @@ def _new_raw_version_index(self, event, oindex):
             return
 
         params = {'upa': upa}
-        extra = {}
         (module, method) = oindex['index_method'].split('.')
         resp = self.mr.run(module, method, params)[0]
         self.mr.cleanup()
+        # TODO: should this case really pass silently?
```
Comment (jayrbolton) [inline IndexRunner/IndexerUtils.py]: :+1: raise or print to error log

File: IndexRunner/IndexerUtils.py (hunk 2 above)
Comment (scanon) [reply IndexRunner/IndexerUtils.py]: Agreed.

File: IndexRunner/IndexerUtils.py [hunk 6]
```diff
@@ -222,13 +227,8 @@ def _get_ws_info(self, wsid):
...
         # Don't index temporary narratives
-        temp = False
-        if meta.get('is_temporary') == 'true':
-            temp = True
-
-        public = False
-        if info[6] != 'n':
-            public = True
+        temp = (meta.get('is_temporary') == 'true')
+        public = (info[6] != 'n')
```
Comment (scanon) [inline IndexRunner/IndexerUtils.py:231]: What about the temp flag?

File: IndexRunner/IndexerUtils.py [hunk 7]
```diff
@@ -237,6 +237,7 @@ def _get_ws_info(self, wsid):
                 'temp': temp, 'public': public, 'shared': shared}
 
     def publish(self, wsid):
+        """TODO: clarify the purpose of this function"""
```
Comment (scanon) [inline IndexRunner/IndexerUtils.py]: This is called to update the state of a workspace that is made public.

File: IndexRunner/IndexerUtils.py [hunk 8]
```diff
@@ -321,14 +322,16 @@ def _update_es_access(self, index, wsid, objid, vers, upa):
         return res
 
     def _split_upa(self, upa):
-        return map(lambda x: int(x), upa.split('/'))
+        return [int(x) for x in upa.split('/')]
 
     def _get_indexes(self, otype):
+        # TODO: handle generics here
```
Comment (scanon) [inline IndexRunner/IndexerUtils.py:323]: Yeah..   We would need some way of specific a wild card on a module level.

File: IndexRunner/IndexerUtils.py [hunk 9]
```diff
@@ -321,14 +322,16 @@ def _update_es_access(self, index, wsid, objid, vers, upa):
...
         if otype in self.mapping:
             return self.mapping[otype]
         return self.mapping['Other']
 
     def _check_mapping(self, oindex, objschema):
+        # TODO: isn't this more "ensure mapping exists" Not super sure of what mapping is for
```
Comment (scanon) [inline IndexRunner/IndexerUtils.py]: True.  The mapping basically the schema in elastic.

File: IndexRunner/IndexerUtils.py (hunk 5 above)
Comment (scanon) [reply IndexRunner/IndexerUtils.py]: Yeah.

File: IndexRunner/IndexerUtils.py [hunk 10]
```diff
@@ -420,9 +430,9 @@ def _new_object_version_multi_index(self, event, oindex):
         self.mr.cleanup()
         parent = extra['parent']
         self._check_mapping(oindex, extra['schema'])
+        # TODO: unused, also can we switch away from the term "features"
```
Comment (scanon) [inline IndexRunner/IndexerUtils.py]: Yeah.  We would need to update the indexers too.

**Review by scanon — CHANGES_REQUESTED** [review]
Great feedback.  Only one thing I wasn't sure about (the dropping of the temp field).

File: IndexRunner/IndexerUtils.py (hunk 4 above)
Comment (JamesJeffryes) [reply IndexRunner/IndexerUtils.py:230]: I know they aren't needed but I think they make the comparison more distinct from the assignment. Could be swayed though

File: IndexRunner/IndexerUtils.py (hunk 6 above)
Comment (JamesJeffryes) [reply IndexRunner/IndexerUtils.py:231]: ? It's still set

**Comment by JamesJeffryes** [comment]
@scanon I implemented several of the changes in this PR but want to avoid blowing it up too much. Once this is merged I have another queued up

**Pull request merged**